syntax = "proto2";

package eflect.protos.sample;

import "protos/sample/battery_manager.proto";
import "protos/sample/jiffies.proto";
import "protos/sample/nvml.proto";
import "protos/sample/rapl.proto";

option java_multiple_files = true;
option java_package = "eflect.protos.sample";

message Sample {
  oneof data {
    // energy data
    BatteryManagerSample battery_manager = 4;
    RaplSample rapl = 1;
    NvmlSample nvml = 5;

    // jiffies data
    CpuSample cpu = 2;
    ProcessSample process = 3;
  }
}

message DataSet {
  // energy data
  repeated BatteryManagerSample battery_manager = 4;
  repeated RaplSample rapl = 1;
  repeated NvmlSample nvml = 5;

  // jiffies data
  repeated CpuSample cpu = 2;
  repeated ProcessSample process = 3;
}
